File: 02_Essential/Structures_and_Functions/main.go
package main

import "fmt"

type Rectangle struct {
	length, breadth int
}

func Initialize(p *Rectangle, l int, b int) {
	(*p).length = l
	(*p).breadth = b
}

func Area(r Rectangle) int {
	return r.length * r.breadth
}

func ChangeLength(p *Rectangle, l int) {
	(*p).length = l
}

func main() {
	r := Rectangle{}
	Initialize(&r, 10, 5)
	Area(r)
	ChangeLength(&r, 20)
	fmt.Printf("%d", Area(r)) //100
	return
}

File: 02_Essential/Structures_and_Functions/main.cpp
#include <stdio.h>
struct Rectangle
{
    int length;
    int breadth;
};
void Initialize (struct Rectangle *p, int l, int b)
{
    p->length = l;
    p->breadth = b;
    // (*p).length = l;
	// (*p).breadth = b;
}
int Area (struct Rectangle r)
{
    return r.length * r.breadth;
}
void ChangeLength (struct Rectangle *p, int l)
{
    p->length = l;
}
int main ()
{
    struct Rectangle r;
    Initialize (&r, 10, 5);
    Area (r);
    ChangeLength (&r, 20);
    printf ("%d", Area(r) ); //100 
}